x/oracle/keeper/band_oracle_test.go

```diff
@@ -274,3 +275,122 @@ func TestGetPrice(t *testing.T) {
+	app := app.Setup(t, false)
+	ctx := app.BaseApp.NewContextLegacy(false, tmproto.Header{Height: 1, ChainID: "3", Time: time.Unix(1618997040, 0)})
+
+	// Define table-driven test cases
+	tests := []struct {
+		name          string
+		clientID      string
+		calldata      *types.CalldataRecord
+		oracleOutput  interface{}
+		expectedError bool
+		expectedRate  int64
+	}{
+		{
+			name:          "Fail when ClientID is not a valid integer",
+			clientID:      "invalid-id", 
+			calldata:      nil,
+			oracleOutput:  nil,
+			expectedError: true,
+		},
+		{
+			name:          "Return nil when no CallDataRecord found",
+			clientID:      "1",
+			calldata:      nil,
+			oracleOutput:  nil,
+			expectedError: false,
+		},
+		{
+			name:     "Fail when decoding OracleInput",
+			clientID: "1",
+			calldata: &types.CalldataRecord{
+				ClientId: 1,
+				Calldata: []byte{0xFF, 0xFF},
+			},
+			oracleOutput:  nil,
+			expectedError: true,
+		},
+		{
+			name:     "Fail when decoding OracleOutput",
+			clientID: "1",
+			calldata: &types.CalldataRecord{
+				ClientId: 1,
+				Calldata: utils.MustEncode(types.Input{
+					Symbols:    []string{"ATOM", "BTC"},
+					Multiplier: types.BandPriceMultiplier,
+				}),
+			},
+			oracleOutput:  []byte{0xFF, 0xFF},
+			expectedError: true,
+		},
+		{
+			name:     "Success with valid OracleResponsePacketData",
+			clientID: "1",
+			calldata: &types.CalldataRecord{
+				ClientId: 1,
+				Calldata: utils.MustEncode(types.Input{
+					Symbols:    []string{"ATOM", "BTC"},
+					Multiplier: types.BandPriceMultiplier,
+				}),
+			},
+			oracleOutput: types.BandOutput{
+				Responses: []types.Response{
+					{Symbol: "ATOM", ResponseCode: 0, Rate: 100},
+					{Symbol: "BTC", ResponseCode: 0, Rate: 50000},
+				},
+			},
+			expectedError: false,
+			expectedRate:  100,
+		},
+	}
+
+	// Iterate over each test case
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.calldata != nil {
+				err := app.OracleKeeper.SetBandCallDataRecord(ctx, tt.calldata)
+				require.NoError(t, err)
+			}
+
+			var result []byte
+			if tt.oracleOutput != nil {
+				switch v := tt.oracleOutput.(type) {
+				case types.BandOutput:
+					result = utils.MustEncode(v)
+				case []byte:
+					result = v
+				}
+			}
+
+			oraclePacketData := types.OracleResponsePacketData{
+				ClientID:    tt.clientID,
+				RequestID:   1,
+				Result:      result,
+				ResolveTime: time.Now().Unix(),
+			}
+
+			relayer := sdk.AccAddress("mock-relayer-address")
+
+			err := app.OracleKeeper.ProcessBandOraclePrices(ctx, relayer, oraclePacketData)
+
+			if tt.expectedError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+
+				record := app.OracleKeeper.GetBandCallDataRecord(ctx, 1)
+				require.Nil(t, record)
+
+				if tt.expectedRate != 0 {
+					priceState := app.OracleKeeper.GetBandPriceState(ctx, "ATOM")
+					require.NotNil(t, priceState)
+					require.Equal(t, tt.expectedRate, priceState.Rate.Int64())
```

Comment:
We should get the `PriceState.Price` and compare instead of rate here

Reply:
We also need to check whether `BandCallDataRecord` was deleted after this line `k.DeleteBandCallDataRecord(ctx, uint64(clientID))`